src/Uno.UWP/UI/Input/GestureRecognizer.Manipulation.cs

```diff
@@ -0,0 +1,283 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Numerics;
+using Windows.Devices.Input;
+using Windows.Foundation;
+using Uno.Extensions;
+
+namespace Windows.UI.Input
+{
+	public partial class GestureRecognizer
+	{
+		private class Manipulation
+		{
+			private enum ManipulationStates
+			{
+				Starting = 1,
+				Started = 2,
+				// Inertia = 3
+				Completed = 4,
+			}
+
+			private readonly GestureRecognizer _recognizer;
+			private readonly PointerDeviceType _deviceType;
+			private readonly GestureSettings _settings;
+
+			private ManipulationStates _state = ManipulationStates.Starting;
+			private Points _origins;
+			private Points _lastPublished;
+			private Points _currents;
+
+			public Manipulation(GestureRecognizer recognizer, PointerPoint pointer1)
+			{
+				_recognizer = recognizer;
+				_settings = recognizer._gestureSettings;
+				_deviceType = pointer1.PointerDevice.PointerDeviceType;
+
+				_origins = _lastPublished = _currents = pointer1;
+
+				_recognizer.ManipulationStarting?.Invoke(_recognizer, EventArgs.Empty);
+			}
+
+			public void Add(PointerPoint point)
+			{
+				if (_state == ManipulationStates.Completed
+					|| point.PointerDevice.PointerDeviceType != _deviceType
+					|| _currents.HasPointer2)
+				{
+					return;
+				}
+
+				_origins.SetPointer2(point);
+				_lastPublished.SetPointer2(point);
+				_currents.SetPointer2(point);
+
+				// As weird it is, it's how UWP behaves: Starting is raised for each pointer
+				_recognizer.ManipulationStarting?.Invoke(_recognizer, EventArgs.Empty);
+
+				// We force to start the manipulation (or update it) a second pointer is pressed
+				NotifyUpdate(forceUpdate: true);
+			}
+
+			public void Update(IList<PointerPoint> updated)
```

Comment:
Using an actual List<T> would be better for performance (because of the enumeration)

Reply:
The IList is coming from the public API of the GestureRecognizer